adagrams/game.py

```diff
@@ -1,11 +1,164 @@
+    Returns 'hand' pointing to a list of drawn letters.
+    '''
+    letter_pool_list = []
+    letter_pool_dict = {}
+    hand = []
+    i = 0
+
+    # append LETTER_POOL keys into a list
+    for key in LETTER_POOL.keys():
+        letter_pool_list.append(key)
+    
+    # copy LETTER_POOL dict
+    for key, value in LETTER_POOL.items():
+        letter_pool_dict[key] = value
+
+    # debug info
+    # print(f"letter_pool is {letter_pool}")
+
+    # random.randint() to append a random index from letter_pool list
+    while i < 10:
+        letter = letter_pool_list[random.randint(0, 25)] 
+
+        # debug info
+        # print(f"letter to check is {letter}")
+        # print(f"letter_pool_dict[letter] is {letter_pool_dict[letter]}")
+
+    # check if letter is available in letter_pool_dict
+        if letter_pool_dict[letter] > 0:
+            hand.append(letter)
+            letter_pool_dict[letter] -= 1
+            i += 1
+
+            # debug info
+            # print(f"hand is {hand}")
+
+    return hand
+    
+hand = draw_letters()
+print(hand)
 
 def uses_available_letters(word, letter_bank):
-    pass
+    '''
+    Checks word uses available letters in letter_bank
+
+    Iterates through each letter in word, checks if letter is available in
+    list copy of letter_bank and removes letter.
+    '''
+    letter_bank_copy = []
+    result = True
+
+    # copy letter_bank
+    for letter in letter_bank:
+        letter_bank_copy.append(letter)
+
+    # iterate through each letter in word, convert to uppercase, and
+    # checks if letter is in letter_bank_copy
+    for letter in word:
+        letter = letter.upper()
+        if letter not in letter_bank_copy:
+            result = False
+            break
+        else:
+            letter_bank_copy.remove(letter)
+    
+    return result
 
 def score_word(word):
-    pass
+    '''Scores word based on values for each letter in word.'''
+    score = 0
+    VALUE_1 = ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"]
+    VALUE_2 = ["D", "G"]
+    VALUE_3 = ["B", "C", "M", "P"]
+    VALUE_4 = ["F", "H", "V", "W", "Y"]
+    VALUE_5 = ["K"]
+    VALUE_8 = ["J", "X"]
+    VALUE_10 = ["Q", "Z"]
+    LEN_FOR_BONUS = [7, 8, 9, 10]
+
+    for letter in word:
+        letter = letter.upper()
+        if letter in VALUE_1:
+            score += 1
+        elif letter in VALUE_2:
+            score += 2
+        elif letter in VALUE_3:
+            score += 3
+        elif letter in VALUE_4:
+            score += 4
+        elif letter in VALUE_5:
+            score += 5
+        elif letter in VALUE_8:
+            score += 8
+        elif letter in VALUE_10:
+            score += 10
+    
+    if len(word) in LEN_FOR_BONUS:
+        score += 8
+    
+    return score
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    '''
+    Compares each word in word_list, returns best_word with highest score.
+    '''
+    best_word = ["", 0]
```

Comment:
Oh, and I just realized, the requirements ask for you to return a tuple. So I should have written

```python
return (best_word, best_word_score)
```

The tests pass because they happen to work whether you return a tuple or a list, but this won't always be true for all languages that distinguish between tuple and list, nor possibly for all test implementations!